src/Compiler/Service/ItemKey.fs

```diff
@@ -181,16 +257,20 @@ and [<Sealed>] ItemKeyStoreBuilder() =
     let writeString (str: string) = b.WriteUTF16 str
 
     let writeRange (m: range) =
+        debug.WriteRange m
         let mutable m = m
         let ptr = &&m |> NativePtr.toNativeInt |> NativePtr.ofNativeInt<byte>
         b.WriteBytes(ptr, sizeof<range>)
 
     let writeEntityRef (eref: EntityRef) =
+        debug.WriteEntityRef eref
         writeString ItemKeyTags.entityRef
         writeString eref.CompiledName
         eref.CompilationPath.MangledPath |> List.iter (fun str -> writeString str)
 
     let rec writeILType (ilTy: ILType) =
+        debug.WriteILType ilTy
```

Comment:
All these probably just should be under `#if DEBUG`, we don't want to even emit those in product.

Reply:
Does it make any difference, when it's just inlined `()` ?

Reply:
Yeah, looks like nothing will be emitted here. 

One extra optimization I'm willing to make is to make a single instance of `DebugKeyStoreNoop` instead of making a new one for each builder.